enterprise/reporting/error_extractor.go

```diff
@@ -345,32 +353,18 @@ func getErrorCodeFromStatTags(statTags map[string]string) string {
 	return strings.Join(errorCodeParts, ":")
 }
 
-func (ext *ExtractorHandle) isVersionDeprecationError(errorMessage string) bool {
-	// Convert to lowercase for case-insensitive matching
-	cleanedError := strings.Replace(strings.ToLower(errorMessage), "-", " ", -1)
-
-	// Check for version-status combinations
-	for _, vPattern := range versionPatterns {
-		if !strings.Contains(cleanedError, vPattern) {
-			continue
-		}
-
-		for _, sPattern := range statusPatterns {
-			if strings.Contains(cleanedError, sPattern) {
-				// Found a version pattern and a status pattern
-				return true
-			}
-		}
-	}
-
-	// Check for strong deprecation terms (using pre-computed list)
-	for _, term := range strongDeprecationTerms {
-		if strings.Contains(cleanedError, term) {
-			return true
-		}
-	}
+func containsDeprecationKey(errorMessage, key string) bool {
+	return strings.HasPrefix(errorMessage, key) || strings.Contains(errorMessage, " "+key)
```

Comment:
Why can't we just search contains? what are the odds that the keyword is a substring of something else?

Reply:
it is quite high, I have come up with only after running against current errors